src/components/application_manager/src/rpc_protection_mediator_impl.cc

```diff
@@ -55,14 +60,33 @@ bool RPCProtectionMediatorImpl::DoesRPCNeedEncryption(
 
   bool encrypted_need = false;
   for (const auto& group : app_rpc_groups) {
-    const bool is_fid_in_group = IsFunctionInGroup(policy_function_id, group);
-    if (is_fid_in_group) {
+    const bool is_function_in_group = IsFunctionInGroup(function_name, group);
+    if (is_function_in_group) {
       encrypted_need |= rpc_encryption_manager.GroupNeedEncryption(group);
     }
   }
+
+  if (encrypted_need) {
+    LOG4CXX_DEBUG(logger_, "Message need encryption");
```

Comment:
i think adding function id or function name would make this log message more meaningful